roles/matrix-nginx-proxy/defaults/main.yml

```diff
@@ -116,8 +116,8 @@ matrix_nginx_proxy_proxy_synapse_enabled: false
 matrix_nginx_proxy_proxy_synapse_hostname: "matrix-nginx-proxy"
 matrix_nginx_proxy_proxy_synapse_federation_api_enabled: "{{ matrix_nginx_proxy_proxy_matrix_federation_api_enabled }}"
 # The addresses where the Federation API is, when using Synapse.
-matrix_nginx_proxy_proxy_synapse_federation_api_addr_with_container: "matrix-synapse:8048"
-matrix_nginx_proxy_proxy_synapse_federation_api_addr_sans_container: "localhost:8048"
+matrix_nginx_proxy_proxy_synapse_federation_api_addr_with_container: "matrix-synapse:{{matrix_synapse_container_default_federation_port|string}}"
```

Comment:
Same. This should not reference `matrix-synapse` variables from within this `matrix-nginx-proxy` role. This should be kept as it is and then redefined in `group_vars/matrix_servers`.

Reply:
So `matrix_nginx_proxy_proxy_synapse_federation_api_addr_with_container` should be empty here too ?

Reply:
It can be. But it doesn't hurt to leave some default value (which doesn't use a variable).